pkg/apis/metering/v1/metering_config.go

```diff
@@ -0,0 +1,489 @@
+	SecretName      string `json:"secretName,omitempty"`
+	CreateSecret    string `json:"createSecret"`
+	AccessKeyID     string `json:"accessKeyID,omitempty"`
+	SecretAccessKey string `json:"secretAccessKey,omitempty"`
+	Endpoint        string `json:"endpoint,omitempty"`
+}
+type PrestoTLS struct {
+	Certificate   string `json:"certificate,omitempty"`
+	Key           string `json:"key,omitempty"`
+	CaCertificate string `json:"caCertificate,omitempty"`
+	CreateSecret  bool   `json:"createSecret"`
+	Enabled       bool   `json:"enabled"`
+	SecretName    string `json:"secretName,omitempty"`
+}
+type PrestoAuth struct {
+	Certificate   string `json:"certificate,omitempty"`
+	Key           string `json:"key,omitempty"`
+	CaCertificate string `json:"caCertificate,omitempty"`
+	CreateSecret  bool   `json:"createSecret"`
+	Enabled       bool   `json:"enabled"`
+	SecretName    string `json:"secretName,omitempty"`
+}
+type PrestoHive struct {
+	HadoopConfigSecretName string `json:"hadoopConfigSecretName,omitempty"`
+	UseHadoopConfig        bool   `json:"useHadoopConfig,omitempty"`
+	MetastoreURI           string `json:"metastoreURI,omitempty"`
+	MetastoreTimeout       string `json:"metastoreTimeout,omitempty"`
+	TLS                    *TLS   `json:"tls,omitempty"`
+}
+type Connectors struct {
+	Hive PrestoHive `json:"hive,omitempty"`
+	// ExtraConnectorFiles []interface{} `json:"extraConnectorFiles,omitempty"`
+}
+type PrestoConfig struct {
+	Environment                     string              `json:"environment,omitempty"`
+	NodeSchedulerIncludeCoordinator bool                `json:"nodeSchedulerIncludeCoordinator,omitempty"`
+	MaxQueryLength                  string              `json:"maxQueryLength,omitempty"`
+	Aws                             *PrestoAWS          `json:"aws,omitempty"`
+	Azure                           *PrestoAzure        `json:"azure,omitempty"`
+	Gcs                             *PrestoGcs          `json:"gcs,omitempty"`
+	S3Compatible                    *PrestoS3Compatible `json:"s3Compatible"`
+	TLS                             *PrestoTLS          `json:"tls,omitempty"`
+	Auth                            *PrestoAuth         `json:"auth,omitempty"`
+	Connectors                      *Connectors         `json:"connectors,omitempty"`
+}
+type MatchExpressions struct {
+	Key      string   `json:"key,omitempty"`
+	Operator string   `json:"operator,omitempty"`
+	Values   []string `json:"values,omitempty"`
+}
+type LabelSelector struct {
+	MatchExpressions []MatchExpressions `json:"matchExpressions,omitempty"`
+}
+type RequiredDuringSchedulingIgnoredDuringExecution struct {
+	LabelSelector *LabelSelector `json:"labelSelector,omitempty"`
+	TopologyKey   string         `json:"topologyKey,omitempty"`
+}
+type PodAntiAffinity struct {
+	RequiredDuringSchedulingIgnoredDuringExecution []RequiredDuringSchedulingIgnoredDuringExecution `json:"requiredDuringSchedulingIgnoredDuringExecution,omitempty"`
+}
+type PrestoAffinity struct {
```

Comment:
No need for this type, you can use https://godoc.org/k8s.io/api/core/v1#Affinity